### DISCUSSION TITLE
Starting eww on a different screen [Hyprland]

### DISCUSSION THREAD

**Discussion by Subbeh** [elkowar/eww#1023] [Q&A]
I'm just going through the example [here](https://elkowar.github.io/eww/configuration.html), but I can't seem to specify the screen to launch eww on using a monitor array.

```
$ eww open example

error: Expected value, but got `Array`
  ┌─ /home/sysadm/.config/eww/eww.yuck:2:21
  │
2 │            :monitor ["<primary>" "LG HDR 4K" "DELL E2720HS" 0]
  │                     ────────────────────────────────────────── Expected some value here
  │
  → Got: Array
```
This is my config:

```
(defwindow example
           :monitor ["<primary>" "LG HDR 4K" "DELL E2720HS" 0]
           :geometry (geometry :x "0%"
                               :y "20px"
                               :width "90%"
                               :height "30px"
                               :anchor "top center")
           :stacking "fg"
           :reserve (struts :distance "40px" :side "top")
           :windowtype "dock"
           :wm-ignore false
  "example content")
```
Also, if I remove the monitor line and run `eww open --screen 1 example`, it will always launch on screen 0 and it takes about 10 seconds to start.

Am I overlooking something?

**Comment by w-lfchen** [comment]
The list functionality seems broken to me as well, using eww-wayland on commit [65d622c](https://github.com/elkowar/eww/commit/65d622c81f2e753f462d23121fa1939b0a84a3e0).
I have not looked into this further, but the screen argument works.
I'd recommend changing your config in the following way:
```
(defwindow example [screen]
           :monitor screen
           :geometry (geometry :x "0%"
                               :y "20px"
                               :width "90%"
                               :height "30px"
                               :anchor "top center")
           :stacking "fg"
           :reserve (struts :distance "40px" :side "top")
           :wm-ignore false
  "example content")
```
This will allow you to specify the screen using the `--screen` option:
`eww open example --screen=0`
where 0 is the ID of the screen you want to use, you can find out the ID using `hyprctl monitors`.
Also note the removal of the `windowtype` line as that is an X11 exclusive option.
